hnn_core/tests/test_network.py

```diff
@@ -250,6 +250,65 @@ def test_network():
         kwargs['probability'] = -1.0
         net.add_connection(**kwargs)
 
+    # Test net.pick_connection()
+    kwargs_default = dict(src_gids=[0, 1], target_gids=[0, 1],
+                          loc=['soma', 'distal'], receptor=['ampa', 'gabaa'])
+
+    # List of tuples with (arg, item, correct_indices)
+    kwargs_good = [
+        ('src_gids', 0, [30, 39]),
+        ('src_gids', 'L2_pyramidal', [29]),
+        ('src_gids', range(2), [30, 39]),
+        ('src_gids', None, [4, 29, 30, 39]),
+        ('target_gids', 35, [22, 34, 35, 37, 38, 40, 41, 42, 43, 44]),
+        ('target_gids', range(2), [30, 39]),
+        ('target_gids', 'L2_pyramidal',
+         [22, 34, 35, 37, 38, 40, 41, 42, 43, 44]),
+        ('target_gids', None,
+         [22, 28, 30, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44]),
+        ('loc', 'soma', [30, 39]),
+        ('loc', None, [30, 39]),
+        ('receptor', 'gabaa', [30, 39]),
+        ('receptor', None, [30, 39])]
```

Comment:
i'm a bit afraid that this hardcoded test will break if something changes the order of the `connectivity` list. Another option is check if all the returned indices indeed contain the match terms and the indices that are not returned don't contain the match term.

Reply:
I agree for most situations, but for this part of the codebase I would really want to know if something changes the order of `net.connectivity`. Fragile tests like 
https://github.com/jonescompneurolab/hnn-core/blob/30fb2a15d02760a7785404b88182683be5b4e455/hnn_core/tests/test_network.py#L256
were really useful for catching that bug in the drives API.

If you forsee a lot of situations where this might crop up and be a pain then I'm totally down to change.